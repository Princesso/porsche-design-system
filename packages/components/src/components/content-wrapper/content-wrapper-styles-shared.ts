import type { JssStyle } from 'jss';
import type { ContentWrapperWidth } from './content-wrapper-utils';
import { getMediaQueryMin, gridSafeZoneBase, gridSafeZoneXXL, gridWidthMax } from '@porsche-design-system/utilities-v2';

const widthMap: { [key in ContentWrapperWidth]?: JssStyle } = {
  basic: {
    margin: '0 auto',
    padding: `0 ${gridSafeZoneBase}`,
    maxWidth: gridWidthMax,
    boxSizing: 'border-box',
    [getMediaQueryMin('xxl')]: {
      padding: `0 ${gridSafeZoneXXL}`,
    },
  },
  extended: {
    margin: '0 auto',
    maxWidth: gridWidthMax,
  },
};

// TODO: maybe this should be part of utilities sub package as "getGridWrapperStyle"?
export const getContentWrapperStyle = (width: ContentWrapperWidth): JssStyle => {
  return {
    width: '100%',
    minWidth: 0, // needed for possible flex context within content-wrapper
    ...widthMap[width],
  };
};
