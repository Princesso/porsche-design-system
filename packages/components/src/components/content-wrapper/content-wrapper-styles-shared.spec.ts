import { getContentWrapperStyle } from './content-wrapper-styles-shared';

describe('getContentWrapperStyle()', () => {
  it.each<Parameters<typeof getContentWrapperStyle>>([['full'], ['fluid'], ['basic'], ['extended'], ['narrow']])(
    'should return correct css for width: %s',
    (...args) => {
      expect(getContentWrapperStyle(...args)).toMatchSnapshot();
    }
  );
});
