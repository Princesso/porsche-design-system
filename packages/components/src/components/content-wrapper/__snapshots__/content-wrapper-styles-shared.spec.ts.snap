// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getContentWrapperStyle() should return correct css for width: basic 1`] = `
{
  "@media(min-width:1920px)": {
    "padding": "0 calc(max(0px, (100% - 2560px) / 2) + clamp(192px, 50vw - 768px, 512px) + clamp(16px, 1.25vw + 12px, 36px) + calc((min(100%, 2560px) - clamp(192px, 50vw - 768px, 512px) * 2 - clamp(16px, 1.25vw + 12px, 36px) * 13) / 14))",
  },
  "@media(min-width:760px)": {
    "padding": "0 calc(clamp(22px, 10.625vw - 12px, 192px) + clamp(16px, 1.25vw + 12px, 36px) + calc((100% - clamp(22px, 10.625vw - 12px, 192px) * 2 - clamp(16px, 1.25vw + 12px, 36px) * 13) / 14))",
  },
  "display": "block",
  "margin": 0,
  "maxWidth": "2560px",
  "minWidth": 0,
  "padding": "0 clamp(22px, 10.625vw - 12px, 192px)",
  "width": "auto",
}
`;

exports[`getContentWrapperStyle() should return correct css for width: extended 1`] = `
{
  "@media(min-width:1920px)": {
    "padding": "0 calc(max(0px, (100% - 2560px) / 2) + clamp(192px, 50vw - 768px, 512px))",
  },
  "display": "block",
  "margin": 0,
  "maxWidth": "2560px",
  "minWidth": 0,
  "padding": "0 clamp(22px, 10.625vw - 12px, 192px)",
  "width": "auto",
}
`;

exports[`getContentWrapperStyle() should return correct css for width: fluid 1`] = `
{
  "display": "block",
  "margin": 0,
  "maxWidth": "2560px",
  "minWidth": 0,
  "padding": "0 max(0px, (100% - 2560px) / 2)",
  "width": "auto",
}
`;

exports[`getContentWrapperStyle() should return correct css for width: full 1`] = `
{
  "display": "block",
  "margin": 0,
  "maxWidth": "2560px",
  "minWidth": 0,
  "padding": "0 max(0px, (100% - 2560px) / 2)",
  "width": "auto",
}
`;

exports[`getContentWrapperStyle() should return correct css for width: narrow 1`] = `
{
  "@media(min-width:1920px)": {
    "padding": "0 calc(max(0px, (100% - 2560px) / 2) + clamp(192px, 50vw - 768px, 512px) + clamp(16px, 1.25vw + 12px, 36px) * 3 + calc((min(100%, 2560px) - clamp(192px, 50vw - 768px, 512px) * 2 - clamp(16px, 1.25vw + 12px, 36px) * 13) / 14) * 3)",
  },
  "@media(min-width:760px)": {
    "padding": "0 calc(clamp(22px, 10.625vw - 12px, 192px) + clamp(16px, 1.25vw + 12px, 36px) * 3 + calc((100% - clamp(22px, 10.625vw - 12px, 192px) * 2 - clamp(16px, 1.25vw + 12px, 36px) * 13) / 14) * 3)",
  },
  "display": "block",
  "margin": 0,
  "maxWidth": "2560px",
  "minWidth": 0,
  "padding": "0 clamp(22px, 10.625vw - 12px, 192px)",
  "width": "auto",
}
`;
